packages/interface-ipfs-core/type/pin/remote.ts

```diff
@@ -0,0 +1,106 @@
+import CID from 'cids'
+import Multiaddr from 'multiaddr'
+import { API as Service } from './remote/service'
+import { AbortOptions } from '../basic'
+
+export interface API {
+  /**
+   * API for configuring remote pinning services.
+   */
+  service: Service
+
+  /**
+   * Stores an IPFS object(s) from a given path to a remote pinning service.
+   */
+  add(cid:CID, options:AddOptions & AbortOptions):Promise<Pin>
+
+  /**
+   * Returns a list of matching pins on the remote pinning service.
+   */
+  ls(query: Query & AbortOptions): AsyncIterable<Pin>
+
+  /**
+   * Removes a single pin object matching query allowing it to be garbage
+   * collected (if needed). Will error if multiple pins mtach provided
+   * query. To remove all matches use `rmAll` instead.
+   */
+  rm(query: Query & AbortOptions): Promise<void>
+
+  /**
+   * Removes all pin object that match given query allowing them to be garbage
+   * collected if needed.
+   */
+  rmAll(query: Query & AbortOptions): Promise<void>
+}
+
+export interface AddOptions extends RemoteServiceOptions {
+  /**
+   * Optional name for pinned data; can be used for lookups later (max 255
+   * characters)
+   */
+  name?: string
+
+  /**
+   * Optional list of multiaddrs known to provide the data (max 20).
+   */
+  origins?: Multiaddr[]
+
+  /**
+   * If true, will add to the queue on the remote service and return
+   * RequestID immediately. If false or amitted will wait until pinned on the
```

Comment:
We don't expose it anymore:
```suggestion
   * immediately. If false or amitted will wait until pinned on the
```